hnn_core/network.py

```diff
@@ -163,53 +156,69 @@ class Network(object):
-        self.p_common, self.p_unique = create_pext(self.params,
-                                                   self.params['tstop'])
-        self.n_common_feeds = len(self.p_common)
-        # Source list of names
-        # in particular order (cells, common, names of unique inputs)
-        self.src_list_new = self._create_src_list()
         # cell position lists, also will give counts: must be known
         # by ALL nodes
-        self.pos_dict = dict.fromkeys(self.src_list_new)
+        # XXX structure of pos_dict determines all downstream inferences of
+        # cell counts, real and artificial
+        self.pos_dict = dict.fromkeys(self.cellname_list)
         self.pos_dict = _create_coords(n_pyr_x=self.params['N_pyr_x'],
                                        n_pyr_y=self.params['N_pyr_y'],
-                                       n_common_feeds=self.n_common_feeds,
-                                       p_unique_keys=self.p_unique.keys(),
                                        zdiff=1307.4)
+
+        # set n_cells, EXCLUDING Artificial ones
         self._count_cells()
 
-        # count external sources
-        self._count_extsrcs()
-        # create dictionary of GIDs according to cell type
-        # global dictionary of gid and cell type
-        self.gid_dict = {}
-        self._create_gid_dict()
+        # Initialise a dictionary of cell ID's, which get assigned when the
+        # network is constructed ('built') in NetworkBuilder
+        # We want it to remain in each Network object, so that the user can
+        # interrogate a built and simulated net. In addition, Spikes are
+        # attached to a Network during simulation---Network is the natural
+        # place to keep this information
+        # XXX rename gid_dict in future
+        self.gid_dict = dict()
+
+        # When computing the network dynamics in parallel, the nodes of the
+        # network (real and artificial cells) potentially get distributed
+        # on different host machines/threads. NetworkBuilder._gid_assign
+        # assigns each node, identified by its unique GID, to one of the
+        # possible hosts/threads for computations. _gid_list here contains
+        # the GIDs of all the nodes assigned to the current host/thread.
+        self._gid_list = []
+
         # Create empty spikes object
         self.spikes = Spikes()
-        # assign gid to hosts, creates list of gids for this node in _gid_list
-        # _gid_list length is number of cells assigned to this id()
-        self._gid_list = []
-        self.trial_idx = 0
+
+        # XXX The legacy code in HNN-GUI _always_ defines 2 'common' and 5
+        # 'unique' feeds. The are added here for backwards compatibility
+        # until a new handling of external NetworkDrives's is completed.
+        self._add_external_feeds()
+
+        # in particular order (cells, common, names of unique inputs)
+        self._update_all_cell_names()
```

Comment:
do you really need this method? It's just one line, why not get rid of it?

actually looking at the code, I see that _all_cell_names is used in only one method? It probably shouldn't even be an attribute then. We should really, really try to strip down the classes to bare minimum as much as possible. Otherwise, code complexity will not go down ...

Reply:
I totally agree, but I'm trying to keep this PR minimally invasive. I'm only 30% done cleaning this stuff up ;)

To answer your questions, I'm not sure it will be needed. But some sort of update-mechanism will be needed when Drives can be added ad hoc, so keeping it as a placeholder for now.